### PR TITLE
Migrate influxd binary to cobra Command package

### LINKED ISSUE #12602: Convert influxd binary to use the cobra CLI package
Future versions of the `influxd` binary will provide tooling via subcommands. To simplify adding these subcommands, migrate to the `cobra` CLI package. 

**TODOs:**

- [x] add `cobra` CLI package
- [x] add `run` subcommand the launches the influxd server
    - [x] If no subcommand is specified when executing `influxd`, the default will be to execute the `run` subcommand. 

----

Some examples of future commands include

* `influxd inspect` - analyze various data files created by InfluxDB
* `influxd gen` - generate data for testing

### PR DESCRIPTION
**Pull request by stuartcarnie**
This PR consists of several improvements or changes to the `influxd` binary:

* migrate the `influxd` binary to the `cobra` CLI package
* introduce a `run` subcommand to start the server
* execute the `run` command by default, if no sub-command is specified
* register the `run` subcommand flags with viper. This ensures compatibility with the existing behavior of automatic binding of flags to environment variables.

Closes #12602

### CODE REVIEW

File: cmd/influxd/launcher/launcher.go [hunk 1]
```diff
@@ -65,6 +68,128 @@ const (
...
+	l := NewLauncher()
+	cmd := &cobra.Command{
+		Use:   "run",
+		Short: "Start the influxd server (default)",
+		Run: func(cmd *cobra.Command, args []string) {
+			// exit with SIGINT and SIGTERM
+			ctx := context.Background()
+			ctx = signals.WithStandardSignals(ctx)
+
+			// m.SetBuild(version, commit, date)
```
Comment (e-dard) [inline cmd/influxd/launcher/launcher.go]: Erroneous comment?

Comment (stuartcarnie) [reply cmd/influxd/launcher/launcher.go]: Yes, let me clean up my mess…

File: cmd/influxd/launcher/launcher.go [hunk 2]
```diff
@@ -65,6 +68,127 @@ const (
...
+	dir, err := fs.InfluxDir()
+	if err != nil {
+		panic(fmt.Errorf("failed to determine influx directory: %v", err))
+	}
+
+	opts := []cli.Opt{
+		{
+			DestP:   &l.logLevel,
+			Flag:    "log-level",
+			Default: "info",
```
Comment (jacobmarble) [inline cmd/influxd/launcher/launcher.go]: I wouldn't mind `zapcore.InfoLevel.String()` but not important.

Comment (stuartcarnie) [reply cmd/influxd/launcher/launcher.go]: Done

**Pull request merged**